### ISSUE TITLE
Querying RedisCollection obtained through generic RedisCollection<T>() fails with InvalidOperationException

### ISSUE THREAD

**Issue by VagyokC4** [redis/redis-om-dotnet#401]
Hi @slorello89.  I found a good one for you.  Hopefully this is a simple fix as having this functionality is critical.

I'm having an issue with querying a RedisCollection obtained through the RedisCollection<T>() method in my data layer while passing in the `<T>`. The same query works fine when I use a RedisCollection that is hardcoded with the `<EntityType>` (not through the methods `<T>`).

Here's a simplified version of my code:
```csharp
public async Task I_Can_Expose_Generic_Issue()
{
    var provider = new RedisConnectionProvider("redis://localhost:6379");
    await provider.Connection.CreateIndexAsync(typeof(CustomerEntity));
    
    var customerEntity       = new CustomerEntity { FirstName = "Test", LastName = "Ing", ERN = "TESTING" };
    var customerWrapper = new GenericWrapper<CustomerEntity>(customerEntity);
    await ExposeGenericIssue( customerWrapper );
}

public async Task ExposeGenericIssue<T>( GenericWrapper<T> request ) where T : class, IRedisEntity
{
    var provider = new RedisConnectionProvider("redis://localhost:6379");
    
    // this works
    var redisCollection1 = provider.RedisCollection<CustomerEntity>(false);
    var working = await redisCollection1.FirstOrDefaultAsync(x => x.ERN == request.Entity.ERN);

    // this does not
    var redisCollection = provider.RedisCollection<T>(false);
    var broken = await redisCollection.FirstOrDefaultAsync(x => x.ERN == request.Entity.ERN);
}

public record GenericWrapper<T>( T Entity ) where T : class, IRedisEntity;

public interface IRedisEntity
{
    string ERN { get; set; }
}

[Document(StorageType = StorageType.Json)]
public class CustomerEntity : IRedisEntity
{
    [Indexed] public                  string   FirstName { get; set; }
    [Indexed] public                  string   LastName  { get; set; }
    public                            string   Email     { get; set; }
    [Indexed(Sortable = true)] public int      Age       { get; set; }
    [Indexed]                  public string[] NickNames { get; set; }

    // IRedisEntity
    [RedisIdField] [Indexed]   public string ERN          { get; set; }
}
```

When I run this, working executes successfully, however broken throws the following error:  

> System.InvalidOperationException: Could not retrieve value from ERN, most likely, it is not properly decorated in the model defining the index.

UPDATE: I believe I found the source of this issue.  Internally you are evaluating my Interface, not the concrete class for the attributes.  Once I place the attributes on the interface as well, all seems to work as expected, for my use case.

```csharp
public interface IRedisEntity
{
    [RedisIdField] [Indexed] string ERN { get; set; }
}
```

**Comment by slorello89** [comment]
Hi @VagyokC4 - I'm trying to think if what this doing is actually wrong, or if there's anything we can do to mitigate it... you're accessing the ERN of `IRedisEntity`, not `CustomerEntity`, so when you get down to what the expression translator is actually translating, it's translating the ERN of `IRedisEntity`, which as you correctly point out, is not decorated:

![image](https://github.com/redis/redis-om-dotnet/assets/42971704/539b5d1a-eff9-4f9e-954f-27e7aace5e77)

It simply doesn't seem to have access to the concrete class at all.

**Comment by VagyokC4** [comment]
> It simply doesn't seem to have access to the concrete class at all.

Yes, that's pretty much where I got to as well.  I couldn't find a concrete way to get at the concrete entity.  Truth be told, I've noticed this behavior with other libraries as well.  e.g. HangFire.NET, where if I want the attributes, and I'm using the interface, I must decorate the Interface.  Once I do that, all works as expected.  We can probably close this out, and hopefully it helps someone in the future, who runs across this.
